### PR TITLE
Docs: Tutorial on adding a Backstage home page

### PR DESCRIPTION
**Pull request by OrkoHunter**
This PR

* Creates a separate page dedicated for Backstage Home page. Currently, if you search for "homepage" on docs, you are unable to find docs on installation.
* Complete installation docs - a lot of the steps were missing in order to get to a working homepage.

### CODE REVIEW

File: docs/getting-started/homepage.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+For App Integrators, the system is designed to be composable to give total freedom in designing a Home Page that suits the needs of the organization. From the perspective of a Component Developer who wishes to contribute with building blocks to be included in Home Pages, there's a convenient interface for bundling the different parts and exporting them with both error boundary and lazy loading handled under the surface.
+
+Let's get started by installing the home plugin and creating a simple home page for your Backstage app.
+
+### Setup home page
+
+#### 1. Install the plugin
+
+```
+# From the Backstage repository root
```
Comment (suuus) [inline docs/getting-started/homepage.md:20]: might be a create-app

Comment (suuus) [inline docs/getting-started/homepage.md:20]: might be a create-app

File: docs/getting-started/homepage.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+cd packages/app
+yarn add @backstage/plugin-home
+```
+
+#### 2. Create a new HomePage component
+
+Inside your `packages/app`, create a new file where our new Home Page component is going to live. Create `packages/app/src/components/home/HomePage.tsx` with the following initial code
```
Comment (suuus) [inline docs/getting-started/homepage.md:27]: +directory

Comment (suuus) [inline docs/getting-started/homepage.md:27]: +directory

File: docs/getting-started/app-custom-theme.md [hunk 3]
```diff
@@ -253,72 +253,3 @@ const LogoFull = () => {
   return <img src={MyCustomLogoFull} />;
 };
 ```
-
-## Custom Homepage
```
Comment (emmaindal) [inline docs/getting-started/app-custom-theme.md:257]: I wonder if its worth keeping this heading and link to the separate page? As I'm wondering if people will miss out on the home page docs when setting up the app if its not findable here? Or at least that was our thinking when putting the docs here in the first place. What do you think?

File: docs/getting-started/homepage.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
+---
+id: homepage
+title: Backstage Home Page - Setup and Customization
```
Comment (emmaindal) [inline docs/getting-started/homepage.md:3]: Niiice! Really happy this got its own page! 🌟

File: docs/getting-started/homepage.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+## Home Page
+
+Having a good Backstage home page can significantly improve the discoverability of the platform. You want your users to find all the things they need right from the home page and never have to remember direct URLs in Backstage. The [Home plugin](https://github.com/backstage/backstage/tree/master/plugins/home) introduces a system for composing a home page for Backstage in order to surface relevant info and provide convenient shortcuts for common tasks. It's designed with composability in mind with an open ecosystem that allows anyone to contribute with any component, to be included in any home page.
+
+For App Integrators, the system is designed to be composable to give total freedom in designing a Home Page that suits the needs of the organization. From the perspective of a Component Developer who wishes to contribute with building blocks to be included in Home Pages, there's a convenient interface for bundling the different parts and exporting them with both error boundary and lazy loading handled under the surface.
+
+Let's get started by installing the home plugin and creating a simple home page for your Backstage app.
+
+### Setup home page
```
Comment (emmaindal) [inline docs/getting-started/homepage.md:15]: Do you think its valuable to mention prerequisites here? (Created app basically)

**Review by emmaindal — COMMENTED** [review]
Couple of initial small comments, in general it looks awesome!

File: docs/getting-started/app-custom-theme.md (hunk 3 above)
Comment (OrkoHunter) [reply docs/getting-started/app-custom-theme.md:257]: Sure, we should keep a line here referencing to customizing homepage. Adding.

File: docs/getting-started/homepage.md (hunk 5 above)
Comment (OrkoHunter) [reply docs/getting-started/homepage.md:15]: Yea, adding pre-reqs and expectations now. :)

File: docs/getting-started/homepage.md (hunk 1 above)
Comment (OrkoHunter) [reply docs/getting-started/homepage.md:20]: Indeed, the "repository" word can confusion between create-app and the mono-repo. Replacing it with "Backstage application directory" as we talked about :)

File: docs/getting-started/homepage.md (hunk 2 above)
Comment (OrkoHunter) [reply docs/getting-started/homepage.md:27]: ✅

File: docs/getting-started/app-custom-theme.md [hunk 6]
```diff
@@ -254,71 +254,7 @@ const LogoFull = () => {
...
-  return (
-    <Grid container spacing={3}>
-      <Grid item xs={12} md={4}>
-        <HomePageCompanyLogo className={container} />
-      </Grid>
-    </Grid>
-  );
-};
-```
+homepage of your app. Read the full guide on the [next page](homepage.md).
```
Comment (freben) [inline docs/getting-started/app-custom-theme.md:260]: Still some inconsistencies of "homepage" vs "home page" :) This seems to be a pretty contended topic! https://english.stackexchange.com/questions/112038/home-page-or-homepage

Comment (emmaindal) [reply docs/getting-started/app-custom-theme.md:260]: Good point @freben when writing the blog post I think we said.

"Home" is the name of the plugin, use that for when referring to the plugin itself.
"homepage" when talking about customising your homepage.
"Homepage Templates" as a product name for the actual templates.

looping in @fengypants here too!

Comment (fengypants) [reply docs/getting-started/app-custom-theme.md:260]: Yep, lotsa spellings! (I think there is also camel case "HomePage" out there.) We decided on "homepage" (one word) and capitalized the way Emma outlined above. 

For example: "As part of the Home plugin for Backstage, you'll find Backstage Homepage Templates, which let you easily customize the look and content of your developer portal's homepage." 

Terrible sentence, but hopefully gets the idea across.

Comment (OrkoHunter) [reply docs/getting-started/app-custom-theme.md:260]: homepage it is!

File: docs/getting-started/homepage.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  </tr>
+</table>
+
+The code for the Backstage sidebar is most likely inside your [`packages/app/src/components/Root/Root.tsx`](https://github.com/backstage/backstage/blob/master/packages/app/src/components/Root/Root.tsx).
+
+Let's make the following changes
+
+```diff
+// Other imports
++ import CategoryIcon from '@material-ui/icons/Category';
```
Comment (andrewthauer) [inline docs/getting-started/homepage.md:103]: I noticed recently there is a `CatalogIcon` exported from `core-components` that might be more appropriate to use here.

That said, there are a few other system icons (eg. `DocsIcon`) that could be used in the demo app and app template as well.

Comment (OrkoHunter) [reply docs/getting-started/homepage.md:103]: Good point @andrewthauer ! I couldn't find the CatalogIcon in the storybook so I didn't know. Maybe a more general topic on Icons consistency 😁

**Review by Rugvip — APPROVED** [review]
🟢 to unblock and you can decide if you want to leave some fixes for later :grin:

File: docs/getting-started/homepage.md (hunk 7 above)
Comment (OrkoHunter) [reply docs/getting-started/homepage.md:103]: cc @katz95 and @emmaindal about the icons 🙏

**Pull request merged**